core/src/main/scala/kafka/log/LogCleanerManager.scala

```diff
@@ -410,7 +410,7 @@ private[log] class LogCleanerManager(val logDirs: Seq[File],
         if (checkpoint != null) {
           val existing = checkpoint.read()
           if (existing.getOrElse(topicPartition, 0L) > offset)
-            checkpoint.write(existing + (topicPartition -> offset))
+            checkpoint.write(mutable.Map() ++= existing += topicPartition -> offset)
```

Comment:
Pardon me, I failed to catch the point of this change.

Reply:
`scala.collection.Map.+` is deprecated.